tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/ShapeOps.java

```diff
@@ -337,4 +336,252 @@
+     */
+    @Endpoint(name = "prepend")
+    public static Operand<TInt64> prepend(
+            Scope scope, Shape<TInt64> shape, long firstDimension) {
+        Operand<TInt64> dim = Constant.arrayOf(scope, new long[]{firstDimension});
+        Concat<TInt64> result = Concat.create(scope,  Arrays.asList(dim, shape),
+                Constant.scalarOf(scope, 0));
+
+        return result;
+    }
+
+
+
+    /**
+     * Returns a new Shape, with another Shape's dimensions prepended.
+     *
+     * @param scope          current scope
+     * @param shape          the TensorFlow shape
+     * @param otherShape     the other shape to prepend
+     * @return a new shape with another Shape's dimensions prepended.
+     */
+    @Endpoint(name = "prepend")
+    public static <T extends TNumber> Operand<T> prepend(
+            Scope scope, Operand<T> shape, Operand<T> otherShape) {
+
+        Concat<T> result = Concat.create(scope,  Arrays.asList(otherShape, shape),
+                Constant.scalarOf(scope, 0));
+
+        return result;
+    }
+
+
+
+    /**
+     * Returns a new Shape, with the last dimension(s) added.
+     *
+     * @param scope         current scope
+     * @param shape         the TensorFlow shape
+     * @param lastDimension the dimension(s) to append
+     * @return a new shape with the shape, followed with the given dimension(s).
+     */
+    @Endpoint(name = "append")
+    public static Operand<TInt32> append(Scope scope, Shape<TInt32> shape, int lastDimension) {
+        Operand<TInt32> dim = Constant.arrayOf(scope, new int[]{lastDimension});
+        Concat<TInt32> result = Concat.create(scope,  Arrays.asList(shape, dim),
+                Constant.scalarOf(scope, 0));
+        return result;
+    }
+
+    /**
+     * Returns a new Shape, with the last dimension(s) added.
+     *
+     * @param scope         current scope
+     * @param shape         the TensorFlow shape
+     * @param lastDimension the dimension(s) to append
+     * @return a new shape with the shape, followed with the given dimension(s).
+     */
+    @Endpoint(name = "append")
+    public static  Operand<TInt64> append(Scope scope, Shape<TInt64> shape, long lastDimension) {
+
+        Operand<TInt64> dim =  Constant.arrayOf(scope, new long[]{lastDimension});
+        Concat<TInt64> result = Concat.create(scope,  Arrays.asList(shape, dim),
+                Constant.scalarOf(scope, 0));
+
+        return result;
+    }
+
+    /**
+     * Returns a new Shape, with another Shape's dimensions prepended.
+     *
+     * @param scope          current scope
+     * @param shape          the TensorFlow shape
+     * @param otherShape     the other shape to prepend
+     * @return a new shape with another Shape's dimensions prepended.
+     */
+    @Endpoint(name = "append")
+    public static <T extends TNumber> Operand<T> append(
+            Scope scope, Operand<T> shape, Operand<T> otherShape) {
```

Comment:
Idem, should the `Operand` here be `Shape` instead?

Reply:
I assume you are referring to org.tensorflow.op.core.Shape.
This class is referring to "dynamic shapes" as detailed in the python code, as opposed to "static shapes" represented
by org.tensorflow.tools.Shape.

org.tensorflow.op.core.Shape is a specific kind of operand, 
but any Operand that translates into a set of numbers works.
For example, tf.constant(new int[] { 1,2,3}), tf.slice(....), tf.concat(), tf.stack(), tf.shape(operand) or tf.constant(org.tensorflow.tools.Shape) all work equally. As far as I can tell there is no way to create a org.tensorflow.op.core.Shape from such arrays of numbers. tf.shape() only takes the shape from an Operand.
The primary use is in creating and manipulating shapes within the Graph operations and outside for Java code per se.

Reply:
closing this thread as it is identical to the one above